source/slang-llvm/slang-llvm-filecheck.cpp

```diff
@@ -0,0 +1,189 @@
+// This file contains a definition of LLVMFileCheck, an implementaion for
+// IFileCheck.
+
+#include <llvm/ADT/SmallString.h>
+#include <llvm/Support/raw_ostream.h>
+#include <llvm/FileCheck/FileCheck.h>
+
+#include <slang.h>
+#include <slang-com-helper.h>
+#include <slang-com-ptr.h>
+#include <core/slang-com-object.h>
+#include <tools/slang-test/filecheck.h>
+
+namespace slang_llvm
+{
+
+using namespace llvm;
+using namespace Slang;
+
+class LLVMFileCheck : IFileCheck, ComBaseObject
+{
+public:
+    // ICastable
+    virtual SLANG_NO_THROW void* SLANG_MCALL castAs(const Guid& guid) override;
+
+    // IUnknown
+    SLANG_COM_BASE_IUNKNOWN_ALL
+    void* getInterface(const Guid& guid);
+    void* getObject(const Guid& guid);
+
+    // IFileCheck
+    virtual TestResult SLANG_MCALL performTest(
+        const char* programName,
+        const char* rulesFilePath,
+        const char* fileCheckPrefix,
+        const CharSlice* stringToCheck,
```

Comment:
Why does this take a pointer to a CharSlice?

Reply:
To save a call to strlen on a potentially large string. Probably a completely trivial cost, and not worth code clarity impact.